common/src/main/java/net/william278/husksync/database/mongo/MongoConnectionHandler.java

```diff
@@ -0,0 +1,70 @@
+package net.william278.husksync.database.mongo;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import lombok.Getter;
+
+import java.util.Collections;
+
+@Getter
+public class MongoConnectionHandler {
+    private final MongoClient mongoClient;
+    private final MongoDatabase database;
+
+    /**
+     * Initiate a connection to a Mongo Server
+     * @param host The IP/Host Name of the Mongo Server
+     * @param port The Port of the Mongo Server
+     * @param username The Username of the user with the appropriate permissions
+     * @param password The Password of the user with the appropriate permissions
+     * @param databaseName The database to use.
+     */
+    public MongoConnectionHandler(String host, int port, String username, String password, String databaseName) {
+        ServerAddress serverAddress = new ServerAddress(host, port);
+        MongoCredential credential = MongoCredential.createCredential(username, databaseName, password.toCharArray());
+
+        MongoClientSettings settings = MongoClientSettings.builder()
+                .credential(credential)
+                .applyToClusterSettings(builder -> builder.hosts(Collections.singletonList(serverAddress)))
+                .build();
+
+        this.mongoClient = MongoClients.create(settings);
+        this.database = mongoClient.getDatabase(databaseName);
```

Comment:
Correct me if I'm wrong, but could we not just have this method call the lower method, passing `authDb` as `null` (with the param annotated `@Nullable` in the lower method) and do a simple null check there to reduce duplicate code?

Reply:
on my phone right now but if this is the method i think it is, i can just remove it as its left over from the copy paste from my mongodb helper